### PR TITLE
Adjust ConvertToKeycardAccount to make it compatible with mobile version

### PR DESCRIPTION
**Pull request by rasom**
https://github.com/status-im/status-go/pull/3145 this PR can be closed 

Two changes here:
- if for some reason account's key can not be deleted the error returned instead of being ignored
- in case if the key can not be decrypted with a new password we attempt to do this with an old one. Currently circumstances at which migration to keycard happens might differ on mobile and desktop versions, so that on mobile there is no guarantee that a user has logged into any account before this operation. Which means configs might be empty and this keys will not be re encrypted on db password change. 

status: ready

### CODE REVIEW

**Comment by ghost** [comment]
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://status.im/docs/))?
- [ ] Have you tested changes with mobile?
- [ ] Have you tested changes with desktop?

File: api/geth_backend.go [hunk 1]
```diff
@@ -564,6 +564,19 @@ func (b *GethStatusBackend) ChangeDatabasePassword(keyUID string, password strin
 	return nil
 }
 
+func (b *GethStatusBackend) tryToDeleteAccount(address types.Address, password, newPassword string) error {
```
Comment (saledjenic) [inline api/geth_backend.go:567]: @rasom once you know about the final requests for mobile app, if the only way to remove convert an account to a keycard account is from within the app (after user logs in) we should remove this and just use `b.accountManager.DeleteAccount` where is needed.

**Review by saledjenic — APPROVED** [review]
@rasom I checked this against the desktop app, it works fine.

File: api/geth_backend.go (hunk 1 above)
Comment (rasom) [reply api/geth_backend.go:567]: yep I understand this, thanks

**Pull request merged**